### PR TITLE
use UTC timestamp for geolocation date

### PR DESCRIPTION
**Pull request by stonebuzz**
### Changes description

Change mysql datetime to timestamp for geolocation date with migration script
to prevent bad date managment
Now twig use server timezone to convert display date  

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@ |1|

### References

Closes #N/A
Related #N/A
Depends on #N/A

### CODE REVIEW

File: tpl/computer_geolocation.html.twig [hunk 1]
```diff
@@ -2,9 +2,9 @@
    <div id="plugin_flyvemdm_geolocationList">
       {% if beginDate is not empty and endDate is not empty %}
          <div><label for="fromDateRange">From date:</label> <span id="fromDateRange"></span>
-         <input type="hidden" id="showdate{{ randBegin }}" value="{{ beginDate|raw }}"/></div>
+         <input type="hidden" id="showdate{{ randBegin }}" value="{{ beginDate|date('Y-m-d H:i:s',timezone)}}"/></div>
          <div><label for="toDateRange">To date:</label> <span id="toDateRange"></span>
-         <input type="hidden" id="showdate{{ randEnd }}" value="{{ endDate|raw }}"/></div>
+         <input type="hidden" id="showdate{{ randEnd }}" value="{{ endDate|date('Y-m-d H:i:s',timezone)}}"/></div>
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply tpl/computer_geolocation.html.twig:7]: it's an input type hidden...

File: tpl/computer_geolocation.html.twig [hunk 2]
```diff
@@ -2,9 +2,9 @@
    <div id="plugin_flyvemdm_geolocationList">
       {% if beginDate is not empty and endDate is not empty %}
          <div><label for="fromDateRange">From date:</label> <span id="fromDateRange"></span>
-         <input type="hidden" id="showdate{{ randBegin }}" value="{{ beginDate|raw }}"/></div>
+         <input type="hidden" id="showdate{{ randBegin }}" value="{{ beginDate|date('Y-m-d H:i:s',timezone)}}"/></div>
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply tpl/computer_geolocation.html.twig:5]: it's an input type hidden

File: inc/geolocation.class.php [hunk 3]
```diff
@@ -206,11 +206,13 @@ private function agentGeolocationInputParser(array $input) {
       $dateGeolocation = false;
       $valuesChecked = false;
       if (isset($input['_datetime'])) {
+
+         $dateGeolocation = new DateTime();
+         $dateGeolocation->setTimestamp($input['_datetime']);
```
Comment (btry) [inline inc/geolocation.class.php]: the 2 lines above duplicate the line with createFromFormat()

File: inc/geolocation.class.php [hunk 4]
```diff
@@ -332,7 +339,9 @@ public static function showForAgent(CommonDBTM $item) {
          'endDate'    => $endDate,
          'randBegin'  => $randBegin,
          'randEnd'    => $randEnd,
-      ];
+         //TODO : GLPI 9.5 retrieve timzeone from user or glpi or server 
+         'timezone'   => date_default_timezone_get(),
+       ];
```
Comment (btry) [inline inc/geolocation.class.php]: an extra space before the bracket

**Review by btry — COMMENTED** [review]
See my comments, minor changes to do IMHO

File: inc/geolocation.class.php (hunk 3 above)
Comment (stonebuzz) [reply inc/geolocation.class.php]: fixed

File: inc/geolocation.class.php (hunk 4 above)
Comment (stonebuzz) [reply inc/geolocation.class.php]: CS fixed

**Comment by stonebuzz** [comment]
AccessLint warning it's an error
see -> https://github.com/AccessLint/accesslint.js/issues/32

**Review by btry — APPROVED** [review]

File: inc/common.class.php [hunk 5]
```diff
@@ -246,4 +246,159 @@ public static function isCurrentUser(PluginFlyvemdmAgent $item) {
...
+            && $this->fieldExists(Config::getTable(), 'value')
+         ) {
+            $conf_tz = $this->request([
+               'SELECT' => 'value',
+               'FROM'   => Config::getTable(),
+               'WHERE'  => [
+                  'context'   => 'core',
+                  'name'      => 'timezone'
+               ]
+            ])->next();
```
Comment (btry) [inline inc/common.class.php]: Why not use 

```php
$config = Config::getConfigurationValues('core', 'timezone');
```

and an extra check with

```php
if (isset($config['timezone')) { ... }
```

**Review by btry — CHANGES_REQUESTED** [review]

File: inc/common.class.php (hunk 5 above)
Comment (stonebuzz) [reply inc/common.class.php]: I simplified the code to find the timezone

**Pull request merged**